src/shell/interpreter.zig

```diff
@@ -10101,6 +10114,734 @@ pub const Interpreter = struct {
+                    .err => |e| {
+                        const buf = switch (e) {
+                            .illegal_option => |opt_str| this.bltn.fmtErrorArena(.cp, "illegal option -- {s}\n", .{opt_str}),
+                            .show_usage => Builtin.Kind.cp.usageString(),
+                            .unsupported => |unsupported| this.bltn.fmtErrorArena(.cp, "unsupported option, please open a GitHub issue -- {s}\n", .{unsupported}),
+                        };
+
+                        _ = this.writeFailingError(buf, 1);
+                        return Maybe(void).success;
+                    },
+                };
+
+                if (maybe_filepath_args == null or maybe_filepath_args.?.len <= 1) {
+                    _ = this.writeFailingError(Builtin.Kind.cp.usageString(), 1);
+                    return Maybe(void).success;
+                }
+
+                const args = maybe_filepath_args orelse unreachable;
+                const paths_to_copy = args[0 .. args.len - 1];
+                const tgt_path = std.mem.span(args[args.len - 1]);
+
+                this.state = .{ .exec = .{
+                    .target_path = tgt_path,
+                    .paths_to_copy = paths_to_copy,
+                } };
+
+                this.next();
+
+                return Maybe(void).success;
+            }
+
+            pub fn ignoreEbusyErrorIfPossible(this: *Cp) void {
+                if (!bun.Environment.isWindows) @compileError("dont call this plz");
+
+                if (this.state.ebusy.idx < this.state.ebusy.state.tasks.items.len) {
+                    outer_loop: for (this.state.ebusy.state.tasks.items[this.state.ebusy.idx..], 0..) |task_, i| {
+                        const task: *ShellCpTask = task_;
+                        assert(task.tgt_absolute != null);
+                        const failure_src = task.src_absolute.?;
+                        const failure_tgt = task.tgt_absolute.?;
+                        if (this.state.ebusy.state.absolute_targets.get(failure_tgt)) |_| {
+                            continue :outer_loop;
+                        }
+                        if (this.state.ebusy.state.absolute_srcs.get(failure_src)) |_| {
+                            continue :outer_loop;
+                        }
+                        this.state.ebusy.idx += i + 1;
+                        this.printShellCpTask(task);
+                        return;
+                    }
+                }
+
+                this.state.ebusy.state.deinit();
+                const exit_code = this.state.ebusy.main_exit_code;
+                this.state = .done;
+                this.bltn.done(exit_code);
+            }
+
+            pub fn next(this: *Cp) void {
+                while (this.state != .done) {
+                    switch (this.state) {
+                        .idle => @panic("Invalid state for \"Cp\": idle, this indicates a bug in Bun. Please file a GitHub issue"),
+                        .exec => {
```

Comment:
maybe there should be a shell panic that includes additional information like what command was run?